crates/primitives/src/snapshot/mod.rs

```diff
@@ -35,4 +38,30 @@ impl HighestSnapshots {
             SnapshotSegment::Receipts => self.receipts,
         }
     }
+
+    /// Returns a mutable reference to a snapshot segment
+    pub fn as_mut(&mut self, segment: SnapshotSegment) -> &mut Option<BlockNumber> {
```

Comment:
see Option::as_mut
```suggestion
    pub fn as_mut(&mut self, segment: SnapshotSegment) -> Option<&mut BlockNumber> {
```

Reply:
but, it might be `None`, and it needs to be assigned to some `Some`